172.md

```diff
@@ -0,0 +1,101 @@
+NIP-172
+=======
+
+Moderated Communities (Reddit Style)
+------------------------------------
+
+`draft` `optional` `author:vitorpamplona` `author:arthurfranca`
+
+The goal of this NIP is to create moderator-approved public communities around a topic. It defines the replaceable event `kind:34550` to define the community and the current list of moderators/administrators. Users that want to post into the community, simply tag any Nostr event with the community's `a` tag (See [NIP-33](33.md)). Moderators issue an approval event `kind:4550` that links the community with the new post.
+
+# Community Definition
+
+`Kind:34550` SHOULD include any field that helps define the community and the set of moderators. `relay` tags MAY be used to describe the preferred relay to download requests and approvals. 
+
+```js
+{
+  "id": "<32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>",
+  "pubkey": "<32-bytes lowercase hex-encoded public key of the event creator>",
+  "created_at": "<Unix timestamp in seconds>",
+  "kind": 34550,
+  "tags": [
+    ["d", "<Community name>"],
+    ["description", "<Community description>"],
+    ["image", "<Community image url>", "<Width>x<Height>"],
+	
+    //.. other tags relevant to defining the community
+
+    // moderators
+    ["p", "<32-bytes hex of a pubkey1>", "<optional recommended relay URL>", "moderator"],
+    ["p", "<32-bytes hex of a pubkey2>", "<optional recommended relay URL>", "moderator"],
+    ["p", "<32-bytes hex of a pubkey3>", "<optional recommended relay URL>", "moderator"],
+	
+    // relays used by the community (w/optional marker)
+    ["relay", "<relay hosting author kind 0>", "author"],
+    ["relay", "<relay where to send and receive requests>", "requests"], 
+    ["relay", "<relay where to send and receive approvals>", "approvals"],
+    ["relay", "<relay where to post requests to and fetch approvals from>"]
```

Comment:
Here, Can we add an 'optional' tag suggestion like this 👇

```javascript
//optional information while forking a community from an existing one
["forksource", "34550:<source community event author pubkey>:<d-identifier of the source community>", "<Optional relay url of source community>"]
```

This would provide a clean mechanism for 
1. Any client to show unmoderated feeds for 'forked communities' based on the user preferences [This is not possible without this additional information].  Client can simply follow the 'forksource' tag and show the events from old community, before the timestamp of forking.
2. Ability to see how many people have forked my community.  This creates a very similar feeling of joy when someone creates a fork of my open-source repo.  This provides motivation for moderators to create a quality community.

Reply:
how do you know the time stamp of the forking since the created at event will change every time the community has a new moderator?

Reply:
Ahhh my bad! Good point. 

How about adding this suggestion instead:

```
["forksource", "34550:<source community event author pubkey>:<d-identifier of the source community>", "<Optional relay url of source community>","optional timestamp of the fork"]
```
This can be considered similar to creating a new git branch from a specific commit